core/state/state_object.go

```diff
@@ -272,10 +272,15 @@ func (s *stateObject) finalise() {
 }
 
 // updateTrie writes cached storage modifications into the object's storage trie.
+// It will return nil if the trie has not been loaded and no changes have been made
+// Note: It may return non-nil if the trie is already loaded due to previous changes
+// in the same block
```

Comment:
I don't see the need for the `Note` section here, just confuses things. The first comment states that it returns `nil` if the trie was no loaded. Implicitly, it will not return `nil` if it was loaded.

Reply:
I kind of agree, it's to clarify that even though there are no effective changes, the trie might still be loaded. Which, I guess, is pretty obvious